template/circle.yml

```diff
@@ -6,7 +6,8 @@ machine:
 
 checkout:
   post:
-    - git fetch origin gh-pages --depth 10
+    - if [ -e .git/shallow ]; then git fetch origin --unshallow; fi
+    - git fetch origin gh-pages
```

Comment:
The problem is, CI was running that code already and was still working with a shallow clone.  The fetch in ghpages.mk doesn't automatically fill in your history, it just gets new stuff.